### DISCUSSION TITLE
--description-template options

### DISCUSSION THREAD

**Discussion by smedir** [RhetTbull/osxphotos#1610] [Q&A] [answered]
Can you have multiple --description-template options in the osxphotos command like --keyword-templates?

I tried to do that but it seems to execute the last --description-template in the command, the ones prior to the last result in no text being added to the description.

**Comment by RhetTbull** [comment, marked as answer]
No, you can have only one description template because the photo can have only one description / caption. Allowing multiple options would be ambiguous -- should they all be concatenated together? What about if there's already a description? Should it be overwritten or concatenated? If concatenating should that be done with newline, comma, space? You can handle all these with the template language in a single template (though they can get a bit complex). For example, the following sets the description to the description followed by a comma and a space then the title if there's both a description and a title otherwise it just outputs the title or the description or nothing of neither is set.

`{descr?{descr},}{descr?{title?{comma} ,},}{title?{title},}`

What exactly are you trying to do with multiple description templates? There's almost certainly a way to do it with a single template.

By the way, `osxphotos template` is a useful tool for testing and trying out templates.

**Reply by smedir** [reply]
> Allowing multiple options would be ambiguous -- should they all be concatenated together?

I think they should be concatenated together if there were such an option.

> If concatenating should that be done with newline, comma, space?

If there is a template being concatenated, it can decide on the terminating concatenation - space, newline, comma.

>  You can handle all these with the template language in a single template (though they can get a bit complex). 

That is quite true and is the crux of the matter- it gets complex very quickly. For two lines, as in your example, it is quite easy. When you have three or more, you get lost in a maze of {} templates and I have not been very successful so far, with multiple errors. If there is a second line - put it in, if there is a fourth line, then you have to check is there a second line/template, third line/template, and then determine the concatenation string and then put in a fourth line etc.

Of course, probably with more skillful templating, and your help :smile: , I might be able to accomplish what I want, however I found the multiple --keyword-template use quite easy and a pleasure to use.

Just my suggestion of course, but thank you for this wonderful tool, I can work around this one way or another with all the other great and comprehensive options.

I also think it would be great to have multiple --title-templates :smile:

**Reply by smedir** [reply]
Your are quite correct, takes a little bit of learning and patience to accomplish, but everything can be accomplished with a properly constructed template.

I was able to export with-

`--description-template "{album?{album}{newline},}{place.name?{place.name|titlecase}{newline},}{title?{title|titlecase}{newline}{newline},}{descr?{descr}{newline},}"
`

**Reply by RhetTbull** [reply]
Great! For now, I do not plan to add the ability to specify multiple options for properties which take a single value for the reasons listed above. Another way to accomplish this is to use a custom `{function:}` template so you can call your own python function. If you already know python that might be easier than the template language.
